MailScanner_perl_scripts/MailWatch.pm

```diff
@@ -384,26 +384,26 @@ sub MailWatchLogging {
     $msg{spamwhitelisted} = $message->{spamwhitelisted};
     $msg{spamblacklisted} = $message->{spamblacklisted};
     $msg{sascore} = $message->{sascore};
-    $msg{spamreport} = $spamreport;
+    $msg{spamreport} = fix_latin($spamreport);
     $msg{ismcp} = $message->{ismcp};
     $msg{ishighmcp} = $message->{ishighmcp};
     $msg{issamcp} = $message->{issamcp};
     $msg{mcpwhitelisted} = $message->{mcpwhitelisted};
     $msg{mcpblacklisted} = $message->{mcpblacklisted};
     $msg{mcpsascore} = $mcpsascore;
-    $msg{mcpreport} = $mcpreport;
+    $msg{mcpreport} = fix_latin($mcpreport);
     $msg{virusinfected} = $message->{virusinfected};
     $msg{nameinfected} = $message->{nameinfected};
     $msg{otherinfected} = $message->{otherinfected};
-    $msg{reports} = $reports;
+    $msg{reports} = fix_latin($reports);
     $msg{hostname} = $hostname;
     $msg{date} = $date;
     $msg{"time"} = $time;
     $msg{headers} = join("\n", map { fix_latin($_)} @{$message->{headers}});
     $msg{quarantined} = $quarantined;
     $msg{rblspamreport} = $message->{rblspamreport};
     $msg{token} = $token;
-    $msg{messageid} = $messageid;
+    $msg{messageid} = fix_latin($messageid);
```

Comment:
Isn't `messageid` limited to ASCII charset?

Reply:
A non-compliant message seems it can have latin chars in here so it bombs out.